### PR TITLE
Migrate `pkg_resources` usages to `importlib.resources`

### PR DESCRIPTION
**Pull request by adferrand**
First part to treat https://github.com/certbot/certbot/issues/9606

In this PR, usages of `pkg_resources` that are now covered by `importlib.resources` are migrated to `importlib`. Backport of `importlib_resources` is added to the dependencies to ensure that the code works on older version of Python (3.7 & 3.8).

The migration essentially follows https://importlib-resources.readthedocs.io/en/latest/migration.html. The code using `pkg_resources.parse_version` are either removed or replaced by `certbot.util.parse_loose_version`.

### CODE REVIEW

File: certbot/certbot/_internal/constants.py [hunk 1]
```diff
@@ -220,8 +226,14 @@
 """Name of the ssl_dhparams file as saved
 in `certbot.configuration.NamespaceConfig.config_dir`."""
 
-SSL_DHPARAMS_SRC = pkg_resources.resource_filename(
-    "certbot", "ssl-dhparams.pem")
+# This code ensures that the resource is accessible as file for the lifetime of current
+# Python process, and will be automatically cleaned up on exit.
+_file_manager = ExitStack()
+atexit.register(_file_manager.close)
+_SSL_DHPARAMS_SRC_REF = importlib_resources.files("certbot") / "ssl-dhparams.pem"
```
Comment (wgreenberg) [inline certbot/certbot/_internal/constants.py:233]: ditto re: placing this in a function

**Review by wgreenberg — CHANGES_REQUESTED** [review]
looks good! just a few suggestions, but feel free to ignore them.

File: certbot-apache/certbot_apache/_internal/constants.py [hunk 2]
```diff
@@ -37,8 +41,11 @@
 ]
 """SHA256 hashes of the contents of previous versions of all versions of MOD_SSL_CONF_SRC"""
 
-AUGEAS_LENS_DIR = pkg_resources.resource_filename(
-    "certbot_apache", os.path.join("_internal", "augeas_lens"))
+_file_manager = ExitStack()
+atexit.register(_file_manager.close)
+_AUGEAS_LENS_DIR_REF = importlib_resources.files("certbot_apache") / "_internal" / "augeas_lens"
+AUGEAS_LENS_DIR = str(_file_manager.enter_context(
+    importlib_resources.as_file(_AUGEAS_LENS_DIR_REF)))
```
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-apache/certbot_apache/_internal/constants.py]: Good idea, I did that!

File: certbot/certbot/_internal/constants.py (hunk 1 above)
Comment (adferrand) [reply certbot/certbot/_internal/constants.py]: Same.

**Review by wgreenberg — APPROVED** [review]

**Comment by wgreenberg** [comment]
LGTM, thanks @adferrand!

**Pull request merged**